sdk/core/azure-core/src/http/curl/curl_session_private.hpp

```diff
@@ -197,7 +197,7 @@ namespace Azure { namespace Core { namespace Http {
      * `STREAMING` and the response has been read completely.
      *
      */
-    SessionState m_sessionState;
+    SessionState m_sessionState = SessionState::PERFORM;
```

Comment:
We were discussing offline whether such private member initializations should occur directly like this, or be reserved to the constructor initialization list or body. I am more familiar with the latter, and that felt more natural to me initially. However, it looks like direct in-class initialization is the recommended pattern and more idiomatic to C++, as long as these member values don't depend on ctor parameters.